src/connect.js

```diff
@@ -191,32 +194,36 @@ export function connectWpPost (contentType, id) {
         return createPostRequest(contentType, realId)
       }
 
-      _makePropsData (query, props) {
+      _makePropsData (props) {
+        const query = this._query()
+
         if (!query || !query.complete || query.error) return null
+
+        const entities = this.props.wordpress.entities[typeConfig.plural]
+        const keys = Object.keys(entities)
         const realId = identifier(displayName, id, props)
-        const entities = this._getState().wordpress.entities[typeConfig.plural]
-        return entities[realId] || null
+
+        for (let i = 0, len = keys.length; i < len; i++) {
```

Comment:
Some content types, e.g. tag, don't have an `id` so for these types we fall back on their `slug`, and vice versa. Maybe there's a better way still?